### DISCUSSION TITLE
Running JDK Mission Control on Apple M1 - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#295] [Announcements]
# Running JDK Mission Control on Apple M1 - Gunnar Morling

JDK Mission Control (JMC) is invaluable for analysing performance data recording using JDK Flight Recorder (JFR).
The other day, I ran into a problem when trying to run JMC on my Mac Mini M1.
Mostly for my own reference, here’s what I did to overcome it.

https://www.morling.dev/blog/running-jmc-on-apple-m1/

**Comment by Thihup** [comment]
`java -Xinternalversion` can also do the trick 🤓

**Reply by gunnarmorling** [reply]
Nice one, TIL! `java -XshowSettings` also works.

**Comment by godofwharf** [comment]
Thanks Gunnar for the article. This command works - `java -XshowSettings 2>&1 | grep "os.arch"`. I got around the same problem by downloading this build of JMC (https://docs.azul.com/azul-mission-control/releases/8.2.0) which supports aarch64.

**Comment by lomoY** [comment]
I downloaded JMC from this build https://adoptopenjdk.net/jmc.html

**Comment by navyxliu** [comment]
Thank you for the information. my problem has been resolved with it. 

On my side, it is opposite.  Vendors do provide arm64 binaries. I can't open them with my x86_64 jdk.  The solution is the same. 
 
```
$file JDK\ Mission\ Control.app/Contents/MacOS/jmc
JDK Mission Control.app/Contents/MacOS/jmc: Mach-O 64-bit executable arm64
```

**Reply by dongshengbc** [reply]
I am having the same problem. Can you share what exactly you did to fix the problem? Thanks in advance.

**Reply by dongshengbc** [reply]
I found the solution. Instead of changing _JDK Mission Control.app/Contents/Info.plist_, I need to update _JDK\ Mission\ Control.app/Contents/Eclipse/jmc.ini_ file. 

----

 cp ~/Downloads/jmc-8.3.1_macos-aarch64/JDK\ Mission\ Control.app/Contents/Eclipse/jmc.ini ~/Downloads/jmc-8.3.1_macos-aarch64/JDK\ Mission\ Control.app/Contents/Eclipse/jmc.ini.back

----

dongsheng@dongsheng-mbp ~ $ /Users/dongsheng
 % command diff ~/Downloads/jmc-8.3.1_macos-aarch64/JDK\ Mission\ Control.app/Contents/Eclipse/jmc.ini ~/Downloads/jmc-8.3.1_macos-aarch64/JDK\ Mission\ Control.app/Contents/Eclipse/jmc.ini.back                              [ruby-2.6.10p210]
8,9d7
< -vm
< /opt/homebrew/Cellar/openjdk/19/bin
dongsheng@dongsheng-mbp ~ $ /Users/dongsheng
 %

**Comment by dhinojosa** [comment]
I got it. The problem on my end was because of sdkman. If you just ported your sdkman settings straight to your M1 in all likelihood you are still using x86 jvm. The reason why the message "libjvm.dylib" appears is because I was using a x86 JDK with and arm JMC hence the error. In fact, every JDK that you have been installing since then has been x86. Here are some of my suggestions.

1.  Uninstall all the JVMs from sdkman because they are probably all x86s and not arm VMs
2. Run `unset SDKMAN_PLATFORM`
3. Open a new terminal and run  `sdk env init` or `sdk env install`
4. Open _~/.sdkman/etc/config_ in an editor
5. Change `sdkman_rosetta2_compatible` to `false` because we _only_ want arm64 SDKs and tools
6. Change `sdkman_debug_mode` to `true` to because this will allow us to view the exact version it is downloading. It's going to be noisy, so you may want to turn it off after you feel confident you have the right versions.
7. Now install some new JDKs and JMCs and all the while check the debugs to ensure everything is ARM64. If there is a difference, in either JDK or JMC then you will get a "libjvm.dylib" errors because they are different architecture libraries.

**Comment by Lexcrd1337** [comment]
You may now download the macOS ARM 64 binary from Oracle: https://www.oracle.com/java/technologies/javase/products-jmc8-downloads.html
